include/seqan3/range/views/kmer_hash.hpp

```diff
@@ -211,6 +211,9 @@ class kmer_hash_view<urng_t>::shape_iterator
     //!\brief The sentinel type of the underlying range.
     using sentinel_t = std::ranges::sentinel_t<rng_t>;
 
+    template <typename urng2_t>
+    friend class shape_iterator;
```

Comment:
I have 2 questions here:
1. Why do we need the friend declaration, although `shape_iterator` is public?
2. Neither in the modified test nor in the iterator test template the `shape_iterator` is used. How do you actually test the new functionality?

Reply:
Good question :)
So below, I introduced a new constructor from a shape_iterator which is specialised over a different `urng_t`. This implies that the two shape_iterators do not have the same type (differnet underlying type).
In order to copy over the members, I need to access them on `rhs`, but they are declared private. That's why I have to befriend every shape_iterator regardless of its underlying `urng_t` type in order to be able to access the private members.

Regarding the tests I answered below. If anything is still not clear please ask again :)

Reply:
Ah thank you, I think I understand now: A class with different template arguments is considered a different class and therefore private and protected member access is prohibited.

Reply:
Yes exactly